src/input.cpp

```diff
@@ -222,26 +248,57 @@ void input_manager::load( const std::string &file_name, bool is_user_preferences
         t_input_event_list events;
         for( const JsonObject keybinding : action.get_array( "bindings" ) ) {
             std::string input_method = keybinding.get_string( "input_method" );
-            input_event new_event;
-            if( input_method == "keyboard" ) {
-                new_event.type = input_event_t::keyboard;
+            std::vector<input_event> new_events( 1 );
+            if( input_method == "keyboard_any" ) {
+                new_events.resize( 2 );
+                new_events[0].type = input_event_t::keyboard_char;
+                new_events[1].type = input_event_t::keyboard_code;
+            } else if( input_method == "keyboard_char" || input_method == "keyboard" ) {
+                new_events[0].type = input_event_t::keyboard_char;
+            } else if( input_method == "keyboard_code" ) {
+                new_events[0].type = input_event_t::keyboard_code;
             } else if( input_method == "gamepad" ) {
-                new_event.type = input_event_t::gamepad;
+                new_events[0].type = input_event_t::gamepad;
             } else if( input_method == "mouse" ) {
-                new_event.type = input_event_t::mouse;
+                new_events[0].type = input_event_t::mouse;
+            } else {
+                keybinding.throw_error( "input_method", "unknown input_method" );
```

Comment:
```suggestion
                keybinding.throw_error( "unknown input_method", "input_method" );
```
The name of the problematic JSON object member is the second parameter.